packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/mapper/ClientMethodsReturnDescription.java

```diff
@@ -0,0 +1,366 @@
+    }
+
+    /**
+     * Creates a {@link reactor.core.publisher.Mono} type wrapping the given type.
+     *
+     * @param type the type to wrap.
+     * @return the Mono type wrapping the given type.
+     */
+    private static IType mono(IType type) {
+        return GenericType.Mono(type);
+    }
+
+    /**
+     * Util method to get the JavaDoc for a client method return type.
+     *
+     * @param operation the operation.
+     * @param returnType the return type of the client method.
+     * @param baseType the base type of the return type.
+     * @return the JavaDoc.
+     */
+    public static String returnTypeJavaDoc(Operation operation, IType returnType, IType baseType) {
+        if (returnType == PrimitiveType.VOID) {
+            // void methods don't have a return value, therefore no return Javadoc.
+            return null;
+        }
+        String description = null;
+        // try the description of the operation
+        if (operation.getLanguage() != null && operation.getLanguage().getDefault() != null) {
+            String operationDescription = SchemaUtil.mergeSummaryWithDescription(operation.getSummary(),
+                operation.getLanguage().getDefault().getDescription());
+            if (!CoreUtils.isNullOrEmpty(operationDescription)) {
+                if (operationDescription.toLowerCase().startsWith("get ")
+                    || operationDescription.toLowerCase().startsWith("gets ")) {
+                    int startIndex = operationDescription.indexOf(" ") + 1;
+                    description = formatReturnTypeJavaDoc(operationDescription.substring(startIndex));
+                }
+            }
+        }
+
+        // try the description on the schema of return type
+        if (description == null && operation.getResponses() != null && !operation.getResponses().isEmpty()) {
+            Schema responseSchema = operation.getResponses().get(0).getSchema();
+            if (responseSchema != null && !CoreUtils.isNullOrEmpty(responseSchema.getSummary())) {
+                description = formatReturnTypeJavaDoc(responseSchema.getSummary());
+            } else if (responseSchema != null
+                && responseSchema.getLanguage() != null
+                && responseSchema.getLanguage().getDefault() != null) {
+                String responseSchemaDescription = responseSchema.getLanguage().getDefault().getDescription();
+                if (!CoreUtils.isNullOrEmpty(responseSchemaDescription)) {
+                    description = formatReturnTypeJavaDoc(responseSchemaDescription);
+                }
+            }
+        }
+
+        if (description == null
+            && baseType == PrimitiveType.BOOLEAN
+            && HttpMethod.HEAD == MethodUtil.getHttpMethod(operation)) {
+            // Mono<Boolean> of HEAD method
+            description = "whether resource exists";
+        }
+
+        description = ReturnTypeDescriptionAssembler.assemble(description, returnType, baseType);
```

Comment:
Would you help rename parameters that contain `description` to `javadoc`, also `ReturnTypeDescriptionAssembler` to `ReturnTypeJavaDocAssembler`(now that `Description` means something different)?

Reply:
Good catch! you're right, it will be confusing to use `description` for java-doc considering the type name. I'll fix it as suggested.